romancal/resample/resample.py

```diff
@@ -0,0 +1,569 @@
+            y dimension varies less rapidly than the x and represents the line
+            index on the input image. If the value is zero, no minimum  will be
+            set in the y dimension.
+
+        ymax : float, optional
+            Sets the maximum value in the y dimension. If the value is zero, no
+            maximum will be set in the y dimension,  the full x dimension
+            of the output image is the bounding box.
+
+        pixfrac : float, optional
+            The fraction of a pixel that the pixel flux is confined to. The
+            default value of 1 has the pixel flux evenly spread across the image.
+            A value of 0.5 confines it to half a pixel in the linear dimension,
+            so the flux is confined to a quarter of the pixel area when the square
+            kernel is used.
+
+        kernel: str, optional
+            The name of the kernel used to combine the input. The choice of
+            kernel controls the distribution of flux over the kernel. The kernel
+            names are: "square", "gaussian", "point", "tophat", "turbo", "lanczos2",
+            and "lanczos3". The square kernel is the default.
+
+        fillval: str, optional
+            The value a pixel is set to in the output if the input image does
+            not overlap it. The default value of INDEF does not set a value.
+
+        Returns
+        -------
+        A tuple with three values: a version string, the number of pixels
+        on the input image that do not overlap the output image, and the
+        number of complete lines on the input image that do not overlap the
+        output input image.
+
+        """
+
+        # Insure that the fillval parameter gets properly interpreted for use with tdriz
+        fillval = "INDEF" if util.is_blank(str(fillval)) else str(fillval)
+        if insci.dtype > np.float32:
+            insci = insci.astype(np.float32)
+
+        # Add input weight image if it was not passed in
+        if inwht is None:
+            inwht = np.ones_like(insci)
+
+        # Compute what plane of the context image this input would
+        # correspond to:
+        planeid = int((uniqid - 1) / 32)
+
+        # Check if the context image has this many planes
+        if outcon.ndim == 2:
+            nplanes = 1
+        elif outcon.ndim == 3:
+            nplanes = outcon.shape[0]
+        else:
+            nplanes = 0
+
+        if nplanes <= planeid:
+            raise IndexError("Not enough planes in drizzle context image")
+
+        # Alias context image to the requested plane if 3d
+        if outcon.ndim == 3:
+            outcon = outcon[planeid]
+
+        if xmin is xmax is ymin is ymax is None:
+            bb = input_wcs.bounding_box
+            ((x1, x2), (y1, y2)) = bb
+            xmin = int(min(x1, x2))
+            ymin = int(min(y1, y2))
+            xmax = int(max(x1, x2))
+            ymax = int(max(y1, y2))
```

Comment:
Now I see why `xmin, xmax, ...` have been dropped from calls to these functions. This code block seems to replicate  JWST's `_resample_range()` but I have questions about this:
1. Is it even possible that `x1 > x2`? (and similar for `y`)
2. [Old code](https://github.com/spacetelescope/jwst/blob/b16e2b78bd601290b633ca74aeaf7383a2c169a5/jwst/resample/resample_utils.py#L315-L319) adds 0.5 pixels before converting `x2` to `int`. That is done to round up so that all input pixels within the bounding box will be resampled. For example, if `x2 = 499.6` then 0.1 (along the `x`-axis) of the 500th pixel has data that could be resampled. With proposed formula you would have excluded that pixel.